projects/gnomad/src/VariantPage/qualityMetrics/GnomadSiteQualityMetrics.js

```diff
@@ -0,0 +1,181 @@
+    super(props)
+
+    this.state = {
+      selectedDataset: props.variant.exome ? 'gnomadExomes' : 'gnomadGenomes',
+      selectedMetric: 'SiteQuality',
+    }
+  }
+
+  render() {
+    return (
+      <AggregateQualityMetricsQuery dataset={this.state.selectedDataset}>
+        {({ data, error, loading }) => {
+          if (loading) {
+            return <p>Loading...</p>
+          }
+          if (error) {
+            return <p>Unable to load metrics</p>
+          }
+
+          const variantData =
+            this.state.selectedDataset === 'gnomadExomes'
+              ? this.props.variant.exome
+              : this.props.variant.genome
+
+          const metricData = data.aggregateQualityMetrics
+
+          let selectedMetricBins
+          let selectedSiteQualityBinDescription
+          if (this.state.selectedMetric === 'SiteQuality') {
+            const variantAlleleFreq = variantData.ac / variantData.an
+
+            if (variantAlleleFreq === 1) {
+              selectedMetricBins = metricData.SiteQuality.singleton.bins
+              selectedSiteQualityBinDescription = 'singleton variants'
+            } else if (variantAlleleFreq === 2) {
+              selectedMetricBins = metricData.SiteQuality.doubleton.bins
+              selectedSiteQualityBinDescription = 'doubleton variants'
+            } else {
+              const selectedAlleleFreqBin = metricData.SiteQuality.af_bins.find(
+                bin => bin.min_af <= variantAlleleFreq && variantAlleleFreq < bin.max_af
+              )
+              selectedMetricBins = selectedAlleleFreqBin.bins
+              selectedSiteQualityBinDescription = `variants with ${
+                selectedAlleleFreqBin.min_af
+              } < AF < ${selectedAlleleFreqBin.max_af}`
+            }
+          } else {
+            selectedMetricBins = metricData[this.state.selectedMetric].bins
+          }
+
+          return (
+            <div>
+              <BarGraph
+                bins={selectedMetricBins}
+                highlightValue={
+                  variantData.qualityMetrics.siteQualityMetrics[this.state.selectedMetric]
+                }
+                logScale={
+                  this.state.selectedMetric === 'SiteQuality' || this.state.selectedMetric === 'DP'
+                }
+                xLabel={this.state.selectedMetric}
+                yLabel="Variants"
+              />
+
+              <ControlSection>
+                <select
+                  onChange={e => {
+                    this.setState({ selectedMetric: e.target.value })
+                  }}
+                  value={this.state.selectedMetric}
+                >
+                  {availableMetrics.map(metric => (
+                    <option key={metric} value={metric}>
+                      {metric} ({variantData.qualityMetrics.siteQualityMetrics[metric]})
+                    </option>
+                  ))}
+                </select>
+
+                <SegmentedControl
```

Comment:
Minor point: site quality metrics plot flickers as you toggle exomes/genomes.

Reply:
It's fetching the overall exome/genome metrics. This can be improved so that it fetches them once up front.